File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-4.cpp

class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        for(auto itr:tokens)
        {   
            char i = itr[0];

            if((i == '+' || i=='-'|| i=='*'||i=='/') && itr.size()==1)
            {
                int res = 0;
                int a1 = st.top();
                st.pop();
                int b1 = st.top();
                st.pop();
                    
                if(i=='+')
                {
                    res=b1+a1;                    
                }
                if(i=='-')
                {
                    res=b1-a1;                    
                }
                if(i=='*')
                {
                    res=b1*a1;                    
                }
                if(i=='/')
                {
                    res=b1/a1;                    
                }
                cout<<res<<endl;
                st.push(res);
                
            }
            else
            {
                st.push(stoi(itr));      
                 
            }
        }
        
        
        return st.top();
        
    }
};

File: Data Structures & Algorithms/minimum-stack/submission-4.go
type MinStack struct {
    q  []int
    mq []int
}

func Constructor() MinStack {
    return MinStack{
        q:  []int{},
        mq: []int{},
    }
}

func (this *MinStack) Push(val int) {
    this.q = append(this.q, val)
    if len(this.mq) == 0 || val <= this.mq[len(this.mq)-1] {
        this.mq = append(this.mq, val)
    }
}

func (this *MinStack) Pop() {
    p := this.q[len(this.q)-1]
    this.q = this.q[:len(this.q)-1]

    if p == this.mq[len(this.mq)-1] {
        this.mq = this.mq[:len(this.mq)-1]
    }
}

func (this *MinStack) Top() int {
    return this.q[len(this.q)-1]
}

func (this *MinStack) GetMin() int {
    return this.mq[len(this.mq)-1]
}